lib/apiservers/engine/backends/container.go

```diff
@@ -415,6 +417,41 @@ func (c *Container) ContainerRm(name string, config *types.ContainerRmConfig) er
 	return nil
 }
 
+// checkActivePortBindings gets port bindings for the container and
+// unmaps ports if the cVM that previously bound them isn't powered on
+func (c *Container) checkActivePortBindings(vc *viccontainer.VicContainer, endpoints []*models.EndpointConfig) error {
+	for p, _ := range vc.HostConfig.PortBindings {
+		thePort := p.Port()
+		if _, mapped := containerByPort[thePort]; mapped {
+			mappedCtr := containerByPort[thePort]
```

Comment:
Why do you read this value twice? you could assign it in if statent where you actually ignoring it.